craft_parts/lifecycle_manager.py

```diff
@@ -65,7 +65,7 @@ def __init__(
         *,
         application_name: str,
         cache_dir: Union[Path, str],
-        work_dir: str = ".",
+        work_dir: Union[Path, str] = ".",
```

Comment:
Maybe to be improved in a next branch: we shouldn't have a duality Path/str for paths. If we want to use Path, let's use Path everywhere. Wherever we need to pass them or receive them as strings, let's convert them as soon as possible, and not leak them as strings in the rest of the code.

Reply:
Will normalize to use Path everywhere in CRAFT-43.